### PR TITLE
docs(tax): DEVDOCS-3835 Add tax properties array to item requests

### PR DESCRIPTION
**Pull request by theromulans**
Document our additive change to allow variable merchant calculation inputs for individual items, via our `tax_properties` field.

This change is already being used by one partner, Wine Direct, and successfully working with our third party tax provider Avalara. It should be safe to publish at any time.

Raised [DEVDOCS-3835](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3835) to describe this change.

ping @bigcommerce/shipping @bc-andreadao

### CODE REVIEW

File: reference/tax_provider.yml
Comment (theromulans) [inline reference/tax_provider.yml:694]: After some further discussion we want to shift this to `code` to clarify it's purpose a little more.

File: reference/tax_provider.yml
Comment (theromulans) [inline reference/tax_provider.yml:985]: @bc-andreadao @slsriehl I noticed a bunch of these have been added more recently, so I've added it here too - does that look correct? Cheers.

**Review by benpratt77 — APPROVED** [review]

File: reference/tax_provider.yml
Comment (bc-andreadao) [reply reference/tax_provider.yml:985]: Yes, we're setting internal flags to false and allowing models to show to avoid schema problems

**Comment by slsriehl** [comment]
@bc-andreadao I set a ticket branch as the target branch, so work with @theromulans and merge this whenever y'all are ready. We can do codeowner review when you open a PR to merge the ticket branch into staging. If Darith is wanting y'all to wait for a CSS release date, we can get this poised to drop from the ticket branch.

**Comment by theromulans** [comment]
We're ready whenever you are @bc-andreadao. Thanks @slsriehl.

**Review by bc-andreadao — APPROVED** [review]

**Comment by slsriehl** [comment]
@bc-andreadao whenever you're ready to complete [DEVDOCS-3835](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3835), open a PR from bigcommerce:DEVDOCS-3835 to bigcommerce:staging.

thanks @theromulans!

**Pull request merged**
